main.go

```diff
@@ -84,48 +88,90 @@ func main() {
 
 	// Make waitgroup and channels to process objects
 	// tasks asynchronously
-	var wg sync.WaitGroup
-	cDone := make(chan bool)
-	wg.Add(len(filtered_res.Contents))
-	go func() {
-		// Process each s3 object
-		for _, item := range filtered_res.Contents {
-			o := item
-			go func() {
-				defer wg.Done()
-				if err := processObject(ctx, s3Cli, dwn, influxWriter, o); err != nil {
-					log.Error().Err(err).
-						Msg("Processing error")
-				}
-			}()
-		}
+	if len(unproc_csvs.Contents) > 0 {
+		var wg sync.WaitGroup
+		cDone := make(chan bool)
+		wg.Add(len(unproc_csvs.Contents))
+		go func() {
+			// Process each s3 object
+			for _, item := range unproc_csvs.Contents {
+				o := item
+				go func() {
+					defer wg.Done()
+					if err := processObject(ctx, dwn, upl, influxWriter, o); err != nil {
+						log.Error().Err(err).
+							Msg("Processing error")
+					}
+				}()
+			}
 
-		wg.Wait()
-		close(cDone)
-	}()
+			wg.Wait()
+			close(cDone)
+		}()
+		<-cDone
+	}
+
+	if opts.CleanObjects && len(proc_csvs.Contents) > 0 {
+		var wgClean sync.WaitGroup
+		cCleanDone := make(chan bool)
+		wgClean.Add(len(proc_csvs.Contents))
+		go func() {
+			// Process each s3 object
+			for _, item := range proc_csvs.Contents {
+				o := item
+				go func() {
+					defer wgClean.Done()
+					if err := cleanObject(ctx, s3Cli, o); err != nil {
+						log.Error().Err(err).
+							Msg("Processing error")
+					}
+				}()
+			}
+
+			wgClean.Wait()
+			close(cCleanDone)
+		}()
+		<-cCleanDone
+	}
 
-	<-cDone
 	log.Info().
 		Dur("elapsed", time.Since(start)).
 		Msg("Processing ended !")
 }
 
-func filterBucketContent(elems s3.ListObjectsOutput, suffix string) (ret s3.ListObjectsOutput) {
+func filterBucketContent(elems s3.ListObjectsOutput, suffix string, processedFlagSuffix string) (unprocessed s3.ListObjectsOutput, processed s3.ListObjectsOutput) {
+	// Rely on .processed files present on the bucket to detect which csv
+	// have already been pushed to influx and which have yet to be processed
+	csvFiles := []types.Object{}
+	processedElems := []string{}
+
 	if len(suffix) == 0 {
-		return elems
+		return unprocessed, processed
 	}
 	for _, s := range elems.Contents {
 		if strings.HasSuffix(*s.Key, suffix) {
-			ret.Contents = append(ret.Contents, s)
+			csvFiles = append(csvFiles, s)
+		}
+		if strings.HasSuffix(*s.Key, processedFlagSuffix) {
+			processedElems = append(processedElems, strings.Replace(*s.Key, processedFlagSuffix, suffix, -1))
```

Comment:
This is equivalent to the following
```suggestion
			processedElems = append(processedElems, strings.ReplaceAll(*s.Key, processedFlagSuffix, suffix))
```